dot/types/babe.go

```diff
@@ -30,17 +30,18 @@ type BabeConfiguration struct {
 	EpochLength        uint64 // duration of epoch in slots
 	C1                 uint64 // (1-(c1/c2)) is the probability of a slot being empty
 	C2                 uint64
-	GenesisAuthorities []*AuthorityRaw
+	GenesisAuthorities []AuthorityRaw
 	Randomness         [RandomnessLength]byte
 	SecondarySlots     byte
 }
 
 // BABEAuthorityRawToAuthority turns a slice of BABE AuthorityRaw into a slice of Authority
-func BABEAuthorityRawToAuthority(adr []*AuthorityRaw) ([]*Authority, error) {
-	ad := make([]*Authority, len(adr))
+func BABEAuthorityRawToAuthority(adr []AuthorityRaw) ([]Authority, error) {
+	ad := make([]Authority, len(adr))
 	for i, r := range adr {
-		ad[i] = new(Authority)
-		err := ad[i].FromRawSr25519(r)
+		ad[i] = Authority{}
+		a := r
+		err := ad[i].FromRawSr25519(&a)
```

Comment:
```suggestion
		err := ad[i].FromRawSr25519(&r)
```

Reply:
@noot This change causes a "Implicit memory aliasing in for loop" am I okay to just add nolint here?

Reply:
weird, ok just leave it I guess lol